cmd/fleet/handleEnroll.go

```diff
@@ -229,6 +235,62 @@ func _enroll(ctx context.Context, bulker bulk.Bulk, c cache.Cache, req EnrollReq
 	return &resp, nil
 }
 
+// updateMetaLocalAgentId updates the agent id in the local metadata if exists
+// At the time of writing the local metadata blob looks something like this
+// {
+//     "elastic": {
+//         "agent": {
+//             "id": "1b9c327a-c93a-4aef-b67f-effbef54d836",
+//             "version": "8.0.0",
+//             "snapshot": false,
+//             "upgradeable": false
+//         }
+//     },
+//     "host": {
+//         "architecture": "x86_64",
+//         "hostname": "eh-Hounddiamond",
+//         "name": "eh-Hounddiamond",
+//         "id": "1b9c327a-c93a-4aef-b67f-effbef54d836"
+//     },
+//     "os": {
+//         "family": "darwin",
+//         "kernel": "19.6.0",
+//         "platform": "darwin",
+//         "version": "10.15.7",
+//         "name": "Mac OS X",
+//         "full": "Mac OS X(10.15.7)"
+//     }
+// }
+func updateLocalMetaAgentId(data []byte, agentId string) ([]byte, error) {
+	if data == nil {
+		return data, nil
+	}
+
+	var m map[string]interface{}
+	err := json.Unmarshal(data, &m)
+	if err != nil {
+		return nil, err
+	}
+
+	if v, ok := m["elastic"]; ok {
```

Comment:
Are we using `common.MapStr` anywhere in fleet-server? If yes, this could be used here with `Get`.

Reply:
We don't, but yeah, could use a better wrapper if we have to do this in any other place.